src/Orchard.Web/Modules/Orchard.MediaLibrary/Services/MediaLibraryService.cs

```diff
@@ -142,12 +145,12 @@ public class MediaLibraryService : IMediaLibraryService {
 
         public string GetUniqueFilename(string folderPath, string filename) {
 
-            // remove any char which is unallowed in an HTTP request
-            foreach (var unallowedChar in HttpUnallowed) {
-                filename = filename.Replace(unallowedChar.ToString(), "");
+            // Removing characters the are forbidden on the file sytem or unallowed in an HTTP request.
```

Comment:
`the are forbidden on the file sytem` => `that are forbidden on the file system`